### DISCUSSION TITLE
Swift 6 support

### DISCUSSION THREAD

**Discussion by Craz1k0ek** [groue/GRDB.swift#1558] [General]
I recently downloaded the new Xcode bèta to try Swift 6 and found that most of GRDB cannot be used out of the box directly, as it's not Swift 6 compatible yet. Do you have any insights or intention to support it? Especially `Sendable` conformance would be a great addition!

**Comment by Jason-Abbott** [comment]
@Craz1k0ek make sure you’ve grabbed the latest. I think `Sendable` conformances were added in [6.26.0](https://github.com/groue/GRDB.swift/releases/tag/v6.26.0).

I am using GRDB extensively with Swift 6 in the new Xcode beta and have no issues (though I may not be using the same types you are).

**Comment by groue** [comment]
@Craz1k0ek, I transformed the issue in a discussion.

The next version GRDB 7 is in the pipes. I opened a lot of issues in the Swift repository so that GRDB can offer a great Swift 6 experience. Some of them are already fixed. Meanwhile, Swift 6 is still beta, and GRDB 7 had not shipped. Please be patient, or [contribute directly](https://github.com/groue/GRDB.swift/blob/master/CONTRIBUTING.md).

Please share your issues with GRDB 6 and Xcode beta, so that we can share workarounds with all GRDB users.

Thank you!

**Comment by Craz1k0ek** [comment]
I'm ashamed to confess that I was mistakenly thinking I was on the latest version. I've been working on multiple projects and one of the two was for sure at 6.27.0, but it didn't cross my mind to check the version on the other one. So sorry. I'll see what testing in the latest bèta is like

**Comment by FledgeXu** [comment]
Hi, I was trying to use GRDB 6.29.0 with swift 6, I just found `BelongsToAssociation` does not support `Sendable` yet?
